.github/scripts/license_years.py

```diff
@@ -0,0 +1,59 @@
+# SPDX-FileCopyrightText: 2024 Google LLC
+#
+# SPDX-License-Identifier: Apache-2.0
+import re
+from git import Repo
+
+# Initialize a git Repo object
+repo = Repo('.')
+git = repo.git
+
+# Get the last common ancestor of the current branch and the staging branch
+current_branch = repo.active_branch
+common_ancestor_hash = repo.git.merge_base('staging', current_branch.name)
+
+# Check if the current branch hash is the same as the staging hash
+if current_branch.commit.hexsha == common_ancestor_hash:
+  print("There are no commits between the current branch and the staging branch")
+  exit()
+
+# Get the commits that are in the current branch but not in main
+all_commits = repo.iter_commits()
+commits = []
+for commit in all_commits:
+  if commit.hexsha == common_ancestor_hash:
+    break
+  commits.append(commit.hexsha)
+
+# Check if the year in the license header is correct for each file in each commit
+all_valid = True
+for commit in commits:
+  obj = repo.commit(commit)
+  commit_year = obj.committed_datetime.year
+  if len(obj.parents) == 0:
+    # Must be an initial commit
+    continue
+  parent = obj.parents[0]
+  # get the diff of the commit
+  diffs = parent.diff(obj)
+  for diff in diffs:
+    # Only check added or modified files
+    if diff.change_type in ['A', 'M']:
+      fp = diff.b_path
+      content = git.show(f"{commit}:{fp}").split('\n')
+      for line in content:
+        # Check if the line contains "SPDX-FileCopyrightText:"
```

Comment:
```suggestion
```

Does what it says on the tin. Comments should either:

* Clarify _why_ things happen
* Or in case of complicated code _what_ happens (but ideally it should be factored out in a function)

In this case, just remove it.